.github/CONTRIBUTING.md

```diff
@@ -86,7 +121,43 @@ The communities behind these PHP frameworks/CMSes/polyfill libraries are strongl
 * Test case files should be placed in the same directory as the unit test file.
 
 
-## Running the Sniff Tests
+## Writing Sniff Tests
+
+Each sniff has its own remit and should function independently of other sniffs.
+
+With that in mind, sniffs are, and should be, tested in isolation.
+The tests for each sniff should only test the functionality of that specific sniff, independently of that the code used to test the sniff may also trigger errors from other sniffs.
+
+Tests for PHPCompatibility sniffs will always consist of sets of two files:
+1. A test case file (the `.inc` file) containing code snippets to run the sniff over.
+    Typically, a test case file should/could contain four closely related categories of code snippets:
+    1. "Not our target" - code which could confuse the sniff, but is not what the sniff is looking for.
+        For example, if a sniff is looking for a function call to a global PHP function, typical "not our target" code snippets would be a function call to a namespaced function with the same name as the global PHP function and method calls to methods with the same name as the global PHP function.
```

Comment:
```suggestion
        For example, if a sniff is looking for a function call to a global PHP function, like `count()`, typical "not our target" code snippets would be a function call to a namespaced function with the same name, like `\Foo\count()`, and calls to methods with the same name, like `$foo->count()`.
```
Added concrete examples to shorten the explanation and make it easier to follow.